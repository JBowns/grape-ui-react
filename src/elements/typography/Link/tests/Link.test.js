import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { MemoryRouter as Router } from 'react-router-dom';
import { Link } from '..';
import 'jest-styled-components';

describe('Link Component base', () => {
  it('should return an anchor tag', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link>Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return an anchor tag with an href, given the href attribute', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link href="./home">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a react router tag, given a \'to\' attribute', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Router><Link to="/home">Lorem Ipsum</Link></Router></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Link Component specified styles', () => {
  it('should return a Link object with a base color (implicit)', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link color="green">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a base color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link color="green.base">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a light color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link color="green.light">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a dark color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link color="green.dark">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with ellipsis', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link ellipsis>Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a specified font-family', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link fontFamily="monospace">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a specified font-weight', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link fontWeight="bold">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with font-style: italic', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link italic>Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with font-size: lg', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link lg>Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with font-size: sm', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link sm>Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a specified kerning', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link kerning="2px">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a specified text-align', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link textAlign="right">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with a specified text-decoration', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link textDecoration="underline">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Link object with custom margin & padding', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Link m={1} p={1} textDecoration="underline">Lorem Ipsum</Link></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
