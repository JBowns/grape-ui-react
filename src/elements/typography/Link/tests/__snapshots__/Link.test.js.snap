// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component base should return a react router tag, given a 'to' attribute 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
  href="/home"
  onClick={[Function]}
>
  Lorem Ipsum
</a>
`;

exports[`Link Component base should return an anchor tag 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component base should return an anchor tag with an href, given the href attribute 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
  href="./home"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a base color (implicit) 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a base color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a dark color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,29.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a light color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,69.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a specified font-family 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: Consolas,"Andale Mono WT","Andale Mono","Lucida Console","Lucida Sans Typewriter","DejaVu Sans Mono","Bitstream Vera Sans Mono","Liberation Mono","Nimbus Mono L",Monaco,"Courier New",Courier,monospace;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a specified font-weight 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: bold;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a specified kerning 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a specified text-align 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: right;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with a specified text-decoration 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with custom margin & padding 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin: 4px;
  padding: 4px;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with ellipsis 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with font-size: lg 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with font-size: sm 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;

exports[`Link Component specified styles should return a Link object with font-style: italic 1`] = `
.c0 {
  background-color: transparent;
  color: hsl(186.8,100%,41.6%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  font-style: italic;
  text-align: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:active {
  color: hsl(207,90%,54%);
  cursor: pointer;
}

<a
  className="c0"
>
  Lorem Ipsum
</a>
`;
